File: calculator-dotnet/Models/CalculationRequest.cs
using System.ComponentModel.DataAnnotations;

namespace CalculatorApp.Models;

/// <summary>Represents a request to perform a calculation.</summary>
/// <param name="FirstNumber">The first operand.</param>
/// <param name="SecondNumber">The second operand.</param>
/// <param name="Operation">The arithmetic operation to perform.</param>
public sealed record CalculationRequest(
    [property: Required] double FirstNumber,
    [property: Required] double SecondNumber,
    [property: Required] Operation Operation
);

File: java-calculator/src/main/java/com/example/calculator/model/CalculationRequest.java
package com.example.calculator.model;

import jakarta.validation.constraints.NotNull;

/**
 * Incoming calculation request.
 *
 * <p>Uses a Java 16+ {@code record} for an immutable, concise data carrier.
 *
 * @param firstNumber  the first operand
 * @param secondNumber the second operand
 * @param operation    the arithmetic operation (ADD, SUBTRACT, MULTIPLY, DIVIDE)
 */
public record CalculationRequest(
        @NotNull(message = "firstNumber must not be null") Double firstNumber,
        @NotNull(message = "secondNumber must not be null") Double secondNumber,
        @NotNull(message = "operation must not be null") String operation) {}
